File: static/java/FormulaBooleanField.java
// ==========================================
// MyAirtable static runtime.
// ==========================================

package myairtable;

/** Formula builder for checkbox/boolean fields. */
public final class FormulaBooleanField implements FormulaField {

  private final String fieldId;

  public FormulaBooleanField(String fieldId) {
    this.fieldId = fieldId;
  }

  @Override
  public String getFieldId() {
    return fieldId;
  }

  /** Equal to boolean value. Named {@code eq} rather than {@code equals} (§2.3.2). */
  public String eq(boolean value) {
    return value ? isTrue() : isFalse();
  }

  /** Field is checked/true. */
  public String isTrue() {
    return field() + "=TRUE()";
  }

  /** Field is unchecked/false. */
  public String isFalse() {
    return field() + "=FALSE()";
  }
}

File: src/myairtable/static/csharp/FormulaBooleanField.cs
namespace MyAirtable;

/// <summary>Formula builder for checkbox/boolean fields.</summary>
public sealed class FormulaBooleanField : FormulaField
{
    public FormulaBooleanField(string fieldId)
        : base(fieldId) { }

    /// <summary>Equal to boolean value. Named <c>Eq</c> rather than <c>Equals</c>.</summary>
    public string Eq(bool value) => value ? IsTrue() : IsFalse();

    /// <summary>Field is checked/true.</summary>
    public string IsTrue() => Field() + "=TRUE()";

    /// <summary>Field is unchecked/false.</summary>
    public string IsFalse() => Field() + "=FALSE()";
}
